### PR TITLE
Solution react_static-list-of-todos

### PR DESCRIPTION
**Pull request by NazariiAlieksieiev**
[DEMO LINK](https://NazariiAlieksieiev.github.io/react_static-list-of-todos/)

### CODE REVIEW

File: src/components/TodoInfo/TodoInfo.tsx [hunk 1]
```diff
@@ -1,4 +1,25 @@
...
-  <>TodoInfo markup</>
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { UserInfo } from '../UserInfo';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoInfo: React.FC<Props> = ({ todo }) => (
```
Comment (loralevitska) [inline src/components/TodoInfo/TodoInfo.tsx:9]: Destructure `todo`

File: src/components/TodoList/TodoList.tsx [hunk 2]
```diff
@@ -1,4 +1,21 @@
...
+
+type Props = {
+  todos: Todo[]
+};
+
 // Add the required types and props
-export const TodoList = () => (
-  <>TodoList markup</>
+export const TodoList: React.FC<Props> = ({ todos = [] }) => (
+  <>
```
Comment (loralevitska) [inline src/components/TodoList/TodoList.tsx:11]: Remove redundant react fragment. https://blog.logrocket.com/understanding-react-fragments/

File: src/components/TodoList/TodoList.tsx [hunk 3]
```diff
@@ -1,4 +1,21 @@
...
+    <section className="TodoList">
+      {todos.map(todo => (
+        <TodoInfo
+          todo={todo}
+          key={todo.id}
+        />
+      ))}
+    </section>
```
Comment (loralevitska) [inline src/components/TodoList/TodoList.tsx:19]: Use semantic tags for list `ul` `li`. Don't forget to remove default `li` dots

File: src/components/UserInfo/UserInfo.tsx [hunk 4]
```diff
@@ -1,4 +1,20 @@
...
+      && (
+        <a
+          key={user.id}
+          className="UserInfo"
+          href={`mailto:${user.email}`}
+        >
+          {user.name}
+        </a>
+      )}
+  </>
```
Comment (loralevitska) [inline src/components/UserInfo/UserInfo.tsx:19]: You check user in TodoInfo
After that check, he will always be here
And destructure `user`
```suggestion
  <a
          key={user.id}
          className="UserInfo"
          href={`mailto:${user.email}`}
        >
          {user.name}
        </a>
```

File: src/components/TodoInfo/TodoInfo.tsx [hunk 5]
```diff
@@ -1,4 +1,25 @@
...
+      className={classNames('TodoInfo', {
+        'TodoInfo--completed': todo.completed,
+      })}
+    >
+      <h2 className="TodoInfo__title">{todo.title}</h2>
+
+      {todo.user
+        && (
+          <UserInfo user={todo.user} />
+        )}
```
Comment (loralevitska) [inline src/components/TodoInfo/TodoInfo.tsx:22]:
```suggestion
      {todo.user && <UserInfo user={todo.user} />}
```

**Review by loralevitska — CHANGES_REQUESTED** [review]
Almost perfect) Let's improve

File: src/components/TodoInfo/TodoInfo.tsx [hunk 6]
```diff
@@ -1,4 +1,28 @@
...
+  todo: Todo;
+};
+
+export const TodoInfo: React.FC<Props> = ({ todo }) => {
+  const {
+    id, completed, title, user,
+  } = todo;
+
+  return (
+    <>
```
Comment (etojeDenys) [inline src/components/TodoInfo/TodoInfo.tsx:15]: also remove this redundant fragment

File: src/components/UserInfo/UserInfo.tsx [hunk 7]
```diff
@@ -1,4 +1,21 @@
...
+
+type Props = {
+  user: User
+};
+
+export const UserInfo: React.FC<Props> = ({ user }) => {
+  const { id, email, name } = user;
+
+  return (
+    <>
```
Comment (etojeDenys) [inline src/components/UserInfo/UserInfo.tsx:11]: redundant fragment

**Review by etojeDenys — CHANGES_REQUESTED** [review]

**Review by VitaliyBondarenko1982 — APPROVED** [review]

**Pull request closed without merging**